### PR TITLE
OpenStack: Parallelize port deletion

### PR DESCRIPTION
**Pull request by dulek**
OpenStack Neutron is slow but it's sometimes able to handle quite a few
requests in parallel. This commit splits the operation of Neutron port
deletion into 10 concurrent goroutines to speed it up. This mostly aids
Kuryr case when we can have hundreds of ports created that block
deletion of other resources.

### CODE REVIEW

**Comment by dulek** [comment]
@MaysaMacedo, @mandre, @pierreprinetti:  Can you please take a look? This seems to speed up `destroy cluster` with Kuryr quite a few times.

**Comment by dulek** [comment]
Without the patch:
> Step e2e-openstack-kuryr-ipi-deprovision-deprovision succeeded after 6m30s. 
With the patch:
> Step e2e-openstack-kuryr-ipi-deprovision-deprovision succeeded after 2m40s.

File: pkg/destroy/openstack/openstack.go [hunk 1]
```diff
@@ -347,49 +347,78 @@ func deletePorts(opts *clientconfig.ClientOpts, filter Filter, logger logrus.Fie
...
+		deletedChannel <- localDeleted
+	}
+
+	deletedChannel := make(chan int)
+	const chunksNumber = 10
+
+	// run processing in chunks
+	for i := 0; i < chunksNumber; i++ {
+		// This is at trick, but it'll produce chunksNumber chunks of roughly the same size.
+		// It may produce empty chunks, but why would we care?
```
Comment (mandre) [inline pkg/destroy/openstack/openstack.go:410]: This doesn't seem right. Take for example `len(allPorts) = 1`, you'd have 10 workers trying to delete the same port, and in the end `numberDeleted` is equal to 10.

You should instead use a buffered channel and run each port delete in its own goroutine.
See https://gobyexample.com/worker-pools for example.

Comment (dulek) [reply pkg/destroy/openstack/openstack.go:410]: This is actually fun, in such case the trick in 411-412 would produce 10 chunks, 9 of which would be empty. Check out and play with [1]. It's suboptimal but I opted to do it that way to make code cleaner.

Now the way you're proposing seems to be even simpler, let me try it out.

[1] https://go.dev/play/p/ogvz3ARDGc9

**Comment by dulek** [comment]
/hold

@mandre proposed alternative approach that seems to be more natural.

File: pkg/destroy/openstack/openstack.go (hunk 1 above)
Comment (pierreprinetti) [reply pkg/destroy/openstack/openstack.go]: Beautiful!
I used this model to test your code: https://go.dev/play/p/6h2w0YgXts3

Comment (dulek) [reply pkg/destroy/openstack/openstack.go]: Right, you've tested the way @mandre proposed, which can also be called "the correct way of doing this". ;)

**Review by mandre — COMMENTED** [review]
Nice, thanks @dulek 
/lgtm

**Comment by dulek** [comment]
> Step e2e-openstack-kuryr-ipi-deprovision-deprovision succeeded after 2m50s. 

Still looks good after latest changes.

**Pull request merged**